src/main/java/htsjdk/samtools/SAMFileWriterFactory.java

```diff
@@ -372,32 +373,37 @@ private SAMFileWriter initWriter(final SAMFileHeader header, final boolean preso
      */
     public SAMFileWriter makeSAMOrBAMWriter(final SAMFileHeader header, final boolean presorted, final File outputFile) {
         final String filename = outputFile.getName();
-        if (filename.endsWith(BamFileIoUtils.BAM_FILE_EXTENSION)) {
-            return makeBAMWriter(header, presorted, outputFile);
-        }
-        if (filename.endsWith(".sam")) {
+        final boolean isSAM = filename.endsWith(SamReader.Type.SAM_TYPE.fileExtension());
+
+        if (isSAM) {
             return makeSAMWriter(header, presorted, outputFile);
+        } else {
+            final boolean isBAM = filename.endsWith(SamReader.Type.SAM_TYPE.fileExtension());
```

Comment:
Shouldn't this be `BAM_TYPE`?  If so, please add a test since obviously this wasn't caught by a test.